crates/dyn-abi/src/parser.rs

```diff
@@ -188,6 +188,7 @@ impl<'a> TryFrom<&'a str> for TupleSpecifier<'a> {
         let value = value.strip_prefix("tuple").unwrap_or(value);
         let value = value.strip_prefix('(').unwrap_or(value);
 
+        // passes over nested tuples
         let mut types = vec![];
         let mut start = 0;
         let mut depth = 0;
```

Comment:
can you explicitly make `depth` `i32` or `isize`, so it doesn't change in the future and cause unwanted overflows?

Reply:
- making this explicitly usize
- handling underflows as mismatched paren errors
- handling mismatched parens more generally
- fixing a bug that allowed mismatched stripping of leading/trailing parents